src/main/java/backend/Logic.java

```diff
@@ -330,9 +330,10 @@ private boolean insertMetadata(Map<Integer, IssueMetadata> metadata, String repo
     }
 
     /**
-     * Carries the current set of GUI elements, as well as the current list of users in the model, to the GUI.
+     * Carries the current set of GUI elements including warning, as well as
+     * the current list of users in the model, to the GUI.
      */
-    public void updateUI(Map<FilterExpression, List<GuiElement>> elementsToShow) {
+    public void updateUI(Map<FilterExpression, ImmutablePair<List<GuiElement>, Optional<String>>> elementsToShow) {
```

Comment:
Tuples are kind of ugly in Java, unfortunately. We should probably make a new class for this.